pandas/tests/test_tseries.py

```diff
@@ -337,6 +337,12 @@ def test_convert_infs():
     assert (result.dtype == np.float64)
 
 
+def test_scientific_no_exponent():
+    arr = np.array(['42E', '2E', '99e', '6e'], dtype='O')
```

Comment:
The changes this PR proposes to the `xstrtod()` functions mean that strings like that are rejected as floats.  The current behaviour converts them as if they were '42E0' etc:

```
In [1]: import pandas as pd, numpy as np

In [2]: pd.__version__
Out[2]: '0.17.1'

In [3]: arr = np.array(['42E', '2E', '99e', '6e'], dtype='O')

In [4]: pd.lib.maybe_convert_numeric(arr, set(), False, True)
Out[4]: array([ 42.,   2.,  99.,   6.])
```

which I believe is incorrect.  The test checks that such strings are rejected as floats, and so give NaN.